hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -210,7 +218,8 @@ public ReplicationManager(final ConfigurationSource conf,
         .addNext(new ClosingContainerHandler(this))
         .addNext(new QuasiClosedContainerHandler(this))
         .addNext(new ClosedWithMismatchedReplicasHandler(this))
-        .addNext(ecReplicationCheckHandler);
+        .addNext(ecReplicationCheckHandler)
+        .addNext(ratisReplicationCheckHandler);
```

Comment:
For the case of EC, the handleChain function would always return false. See line 465 of this file
boolean handled = containerCheckChain.handleChain(checkRequest);
    if (!handled) {
      LOG.debug("Container {} had no actions after passing through the " +
          "check chain", containerInfo.containerID());
    }
This would be always logged in debug mode which is wrong.

Reply:
I guess we should have a Unit Test case for checking this for both EC & Ratis if it is not there.

Reply:
> For the case of EC, the handleChain function would always return false.

I didn't understand this part. If the EC handler returns true, then `handleChain` will also return true. Right?